### PR TITLE
Allow -o with multiple files

### PR DESCRIPTION
**Pull request by terrelln**
Allow `-o` with multiple files, which will work like `-c`.

### CODE REVIEW

**Comment by Cyan4973** [comment]
tests seems to be hitting an unrelated limitation on Windows.

Also, unrelated to this change but present in test traces,
the following error message is likely incorrect : 
https://ci.appveyor.com/project/YannCollet/zstd-p0yf0/build/1.0.1994/job/7d127hm57g57x7lc#L1617

File: tests/playTests.sh [hunk 1]
```diff
@@ -165,6 +165,22 @@ rm tmp
 $ZSTD -f tmp && die "tmp not present : should have failed"
 test ! -f tmp.zst  # tmp.zst should not be created
 
+$ECHO "test : compress multiple files"
+echo hello > tmp1
+echo world > tmp2
+$ZSTD tmp1 tmp2 -o "$INTOVOID"
```
Comment (Cyan4973) [inline tests/playTests.sh]: Is this test supposed to work ?
I was expecting it to fail, because there are 2 input files associated with `-o` command.

Comment (terrelln) [reply tests/playTests.sh]: It only works with `/dev/null` or `-c`.

**Comment by terrelln** [comment]
Rebase on top of PR #949 to fix the Windows test, since `$INTOVOID == NUL != nulmark == nul`.

**Pull request merged**